### PR TITLE
build tests of the tableflip command then improve features

### PR DESCRIPTION
**Pull request by elstamey**
- [x] build a first test of the tableflip command
- [x] test and fix non-alphanumeric characters in the plugin
  This should resolve issue #1when successfully completed

### CODE REVIEW

File: tests/PluginTest.php [hunk 1]
```diff
@@ -35,8 +35,28 @@ public function testGetSubscribedEvents()
...
+        Phake::when($event)->getCommand()->thenReturn('PRIVMSG');
+        $queue = $this->getMockEventQueue();
+
+        $plugin = new Plugin;
+        $plugin->handleTableflipCommand($event, $queue);
+
+        Phake::when($event)->getCustomParams()->thenReturn(array('#channel1', 'tableflip some text'));
+        $plugin->handleTableflipCommand($event, $queue);
+
+        Phake::inOrder(
```
Comment (elstamey) [inline tests/PluginTest.php]: There was 1 failure:

1) Phergie\Irc\Tests\Plugin\React\TableFlip\PluginTest::testHandleTableflipCommand
Failed asserting that Expected Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(equal to string:#channel1, equal to <array>) to be called at least <1> times, actually called <0> times.
Other Invocations:
  Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(string:#channel1, <null>).

/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/vendor/phake/phake/src/Phake/Client/PHPUnit.php:56
/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/vendor/phake/phake/src/Phake/Proxies/VerifierProxy.php:109
/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/tests/PluginTest.php:56
/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/tests/PluginTest.php:56

FAILURES!
Tests: 3, Assertions: 3, Failures: 1.

Comment (ghost) [inline tests/PluginTest.php]: try this for line 56

Phake::verify($queue, Phake::atLeast(1))->ircPrivmsg('#channel1', isString('tableflip some text'))

File: tests/PluginTest.php [hunk 2]
```diff
@@ -35,8 +35,30 @@ public function testGetSubscribedEvents()
...
+
+        $plugin->handleTableflipCommand($event, $queue);
+        Phake::when($event)->getCustomParams()->thenReturn(array($channels));
+
+        $plugin->handleTableflipCommand($event, $queue);
+        Phake::when($event)->getCustomParams()->thenReturn(array($channels,'!tableflip','I','literally','just','ate','a','cricket','right','now'));
+
+        Phake::inOrder(
+            Phake::verify($queue, Phake::atLeast(1))->ircPrivmsg('#channel1', $this->isType('string')),
+            Phake::verify($queue)->ircPrivmsg('#channel1', '(╯°□°）╯︵ ┻━┻  ʍou ʇɥƃıɹ ʇǝʞɔıɹɔ ɐ ǝʇɐ ʇsnɾ ʎꞁꞁɐɹǝʇıꞁ I')
```
Comment (elstamey) [inline tests/PluginTest.php]: here was 1 failure:

1) Phergie\Irc\Tests\Plugin\React\TableFlip\PluginTest::testHandleTableflipCommand
Failed asserting that Expected Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(equal to string:#channel1, equal to <string:(╯°□°）╯︵ ┻━┻  ʍou ʇɥƃıɹ ʇǝʞɔıɹɔ ɐ ǝʇɐ ʇsnɾ ʎꞁꞁɐɹǝʇıꞁ I>) to be called exactly <1> times, actually called <0> times.
Other Invocations:
  Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(string:#channel1, <string:Usage: tableflip message>)
  Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(string:#channel1, <string:message - the message to flip and then send (all words after this are assumed to be part of message)>)
  Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(string:#channel1, <string:Instructs the bot to return a flipped table and the inverted words in the message.>)
  Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(string:#channel1, <string:(╯°□°）╯︵ ┻━┻ >)
  Phergie\Irc\Bot\React\EventQueueInterface->ircPrivmsg(string:#channel1, <null>).

/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/vendor/phake/phake/src/Phake/Client/PHPUnit.php:56
/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/vendor/phake/phake/src/Phake/Proxies/VerifierProxy.php:109
/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/tests/PluginTest.php:59
/var/www/html/phergie/phergie/scaffold/phergie-irc-plugin-react-tableflip/tests/PluginTest.php:59

**Comment by elstamey** [comment]
@matthewtrask @svpernova09 @vreemt @elazar

I am flipping characters like pancakes!  Have you got any requests for additional flips?  Otherwise I'm ready to merge!

**Comment by svpernova09** [comment]
:+1: from here, tests pass locally. Have you tried it in your own instance of phergie?

File: src/Plugin.php [hunk 3]
```diff
@@ -219,4 +233,48 @@ private function utf8_chr($cp) {
...
+     * Switch statement to flip the special and punctuation characters
+     *
+     * @param string $char
+     *
+     * @return string
+     */
+    private function special_char($char)
+    {
+        switch($char) {
+            case "!":
```
Comment (PSchwisow) [inline src/Plugin.php:250]: Why not just add these characters to the array?

File: src/Plugin.php [hunk 4]
```diff
@@ -219,4 +233,48 @@ private function utf8_chr($cp) {
...
     }
+
+    /**
+     * Switch statement to flip the special and punctuation characters
+     *
+     * @param string $char
+     *
+     * @return string
+     */
+    private function special_char($char)
```
Comment (PSchwisow) [inline src/Plugin.php]: PSR-1 section 4.3 states: "Method names MUST be declared in camelCase()."

(Please correct other methods as well.)

**Comment by PSchwisow** [comment]
Other style issues on pre-existing lines (see https://github.com/phergie/phergie-irc-bot-react/blob/master/CONTRIBUTING.md):
1. "Methods, Properties, and Variables should be camelCase"
2. "Short array syntax"

File: src/Plugin.php (hunk 3 above)
Comment (elstamey) [inline src/Plugin.php:250]: array[!] ?

File: src/Plugin.php (hunk 4 above)
Comment (elstamey) [inline src/Plugin.php]: old habits die hard, ya know?

File: src/Plugin.php (hunk 3 above)
Comment (PSchwisow) [inline src/Plugin.php:250]: It's a valid string (if you include the quotes)... it can be an array key.

File: src/Plugin.php (hunk 4 above)
Comment (PSchwisow) [inline src/Plugin.php]: Then I will help you kill them!

Comment (elstamey) [inline src/Plugin.php]: It mostly creeps in with these types of scripty things

**Comment by vreemt** [comment]
awesome, thanks for the fix (:
also, to all those that helped here - thanks for the effort of writing tests and implementing/following coding standards

**Comment by elstamey** [comment]
:+1:

**Pull request merged**